Tests/Command/CleanMediaCommandTest.php

```diff
@@ -276,8 +288,58 @@ public function testExecute()
 
         $output = $this->tester->execute(array('command' => $this->command->getName()));
 
-        $this->assertRegExp('@Context: foo\s+\'qwertz.ext\' was successfully removed\s+\'thumb_1_bar.ext\' was successfully removed\s+done!@', $this->tester->getDisplay());
-
+        $this->assertOutputFoundInContext(
+            '/Context: foo\s+(.+)\s+done!/ms',
+            array(
+                '\'qwertz.ext\' was successfully removed',
+                '\'thumb_1_bar.ext\' was successfully removed',
+            ),
+            $this->tester->getDisplay()
+        );
         $this->assertSame(0, $output);
     }
+
+    /**
+     * Asserts whether all expected texts can be found in the output within a given context.
+     *
+     * @param string $extract  PCRE regex expected to have a single matching group, extracting the content of a context
+     * @param array  $expected Excerpts of text expected to be found in the output
+     * @param string $output   Searched output
+     */
+    private function assertOutputFoundInContext($extractor, $expected, $output)
+    {
+        preg_match_all($extractor, $output, $matches);
+
+        $found = false;
+        foreach ($matches[1] as $match) {
+            if ($this->doesContainAll($match, $expected)) {
+                $found = true;
+                break;
+            }
+        }
+
+        $this->assertTrue($found, sprintf(
+            'Unable to find "%s" in "%s" with extractor "%s"',
+            implode('", "', $expected),
+            $output,
+            $extractor
+        ));
+    }
+
+    /**
+     * Returns whether every needle can be found as a substring of the haystack.
+     *
+     * @param string $haystack
+     * @param array  $needles  Array of (potential) substrings of the haystack
+     */
+    private function doesContainAll($haystack, $needles)
```

Comment:
"doesContainAll" => "containsAll", or do you really want to insist on this fact, like "Yes, it **does** contain all parts" (not very likely, but…) ? Otherwise it easier to understand like this, thanks!